src/Microsoft.Data.SqlClient/tests/ManualTests/SQL/JsonTest/JsonTest.cs

```diff
@@ -0,0 +1,157 @@
+                    using (SqlCommand command2 = connection.CreateCommand())
+                    {
+                        command2.CommandText = spName;
+                        command2.CommandType = CommandType.StoredProcedure;
+                        var parameter2 = new SqlParameter("@jsonData", SqlDbTypeExtensions.Json);
+                        parameter2.Value = jsonDataString;
+                        command2.Parameters.Add(parameter2);
+                        int rowsAffected3 = command2.ExecuteNonQuery();
+                        _output.WriteLine($"Rows affected: {rowsAffected3}");
+                    }
+                }
+            }
+        }
+
+        [ConditionalFact(typeof(DataTestUtility), nameof(DataTestUtility.IsJsonSupported))]
+        public void TestJsonRead()
+        {
+            string jsonDataString = "[{\"name\":\"Dave\",\"skills\":[\"Python\"]},{\"name\":\"Ron\",\"surname\":\"Peter\"}]";
+
+            string tableName = DataTestUtility.GetUniqueNameForSqlServer("Json_Test");
+            string spName = DataTestUtility.GetUniqueNameForSqlServer("spJson_ReadTest");
+
+            string tableCreate = "CREATE TABLE " + tableName + " (Data json)";
+            string tableInsert = "INSERT INTO " + tableName + " VALUES (@jsonData)";
+            string tableRead = "SELECT * FROM " + tableName;
+            string spCreate = "CREATE PROCEDURE " + spName + "AS " + tableRead;
+
+            using (SqlConnection connection = new SqlConnection(DataTestUtility.TCPConnectionString))
+            {
+                connection.Open();
+                using (SqlCommand command = connection.CreateCommand())
+                {
+                    //Create Table
+                    command.CommandText = tableCreate;
+                    command.ExecuteNonQuery();
+
+                    //Create SP for reading from json column
+                    command.CommandText = spCreate;
+                    command.ExecuteNonQuery();
+
+                    //Insert sample json data
+                    //This will be used for reading
+                    command.CommandText = tableInsert;
+                    var parameter = new SqlParameter("@jsonData", SqlDbTypeExtensions.Json);
+                    parameter.Value = jsonDataString;
+                    command.Parameters.Add(parameter);
+                    command.ExecuteNonQuery();
+
+                    //Test 1
+                    //Read json value using query
+                    command.CommandText = tableRead;
+                    SqlDataReader reader = command.ExecuteReader();
+                    while (reader.Read())
+                    {
+                        string jsonData = reader.GetString(0);
+                        _output.WriteLine(jsonData);
+                        Assert.Equal(jsonDataString,jsonData);
+                    }
+
+                    //Test 2
+                    //Read the column metadata
```

Comment:
I'd recommend adding multiple tests instead of running all tests in 1 big one. You can create tables in setup and drop them in tear down.

Reply:
These are quite small tests (single operation) and uses common sql command. Should I break them into individual tests?

Reply:
I think it makes sense to keep this test as is. Data is being read and various aspects of the data are being verified, i.e. Data / Metadata.

However it would be nicer to see these refactored into various methods.
e.g. You could have a reader being executed and data read/asserted in a different function, and then from the same reader extract the Schema and send it to another function to validate. 

Having multiple tests also tend to increase runtime, for setup which, could otherwise have been done in a single test. There is no golden rule. But for this test, it makes sense to have a single `ConditionalFact` and then have multiple verification methods being called from the same test.

Basically 
```
var reader = command.ExecuteReader()
ValidateRows(reader);
ValidateScehma(reader.GetColumnSchema());
reader.Close();
```

Reply:
Refactored test cases.